core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -414,8 +415,9 @@ private synchronized IndexService createIndexService(final String reason, final
     /**
      * This method verifies that the given {@link IndexMetaData} holds sane values to create an {@link IndexService}. This method will throw an
      * exception if the creation fails. The created {@link IndexService} will not be registered and will be closed immediately.
+     * The given {@link Consumer} is invoked with the created {@link IndexService} before closure.
      */
-    public synchronized void verifyIndexMetadata(final NodeServicesProvider nodeServicesProvider, IndexMetaData metaData) throws IOException {
+    public synchronized void verifyIndexMetadata(final NodeServicesProvider nodeServicesProvider, IndexMetaData metaData, Consumer<IndexService> validator) throws IOException {
```

Comment:
I think we should maybe make it even simpler and add this signature:
```Java
public synchronized void verifyIndexMetadata(final NodeServicesProvider nodeServicesProvider, IndexMetaData metaData, IndexMetaData metaDataUpdate) throws IOException {```

and just call `IndexService#updateMetadata()` if it's not the same?